pkg/agent/client/client.go

```diff
@@ -109,17 +111,26 @@ func (c *client) FetchUpdates(ctx context.Context, req *node.FetchX509SVIDReques
 			break
 		}
 		if err != nil {
-			logrus.Errorf("failed to consume entire SVID update stream: %v", err)
+			c.c.Log.Errorf("failed to consume entire SVID update stream: %v", err)
 			c.release(nodeConn)
 			return nil, err
 		}
 
 		if resp.SvidUpdate == nil {
-			logrus.Warn("empty update in SVID update stream")
+			c.c.Log.Warn("empty update in SVID update stream")
 			continue
 		}
 
 		for _, re := range resp.SvidUpdate.RegistrationEntries {
+			if err := validateRegistrationEntry(re); err != nil {
+				c.c.Log.WithFields(logrus.Fields{
+					telemetry.RegistrationID: re.EntryId,
+					telemetry.SPIFFEID:       re.SpiffeId,
+					telemetry.Selectors:      strconv.Itoa(len(re.Selectors)),
+					telemetry.Error:          err.Error(),
+				}).Warn("Entry failed validation")
```

Comment:
nit: an error message that indicates that this bad entry was received from the server would probably help in troubleshooting if/when encountered e.g. "Received malformed registration entry from SPIRE server"

Reply:
:+1